src/formatDate.js

```diff
@@ -9,6 +9,34 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const seperator = date.split(fromFormat[3]);
+  let formattedDate = [];
+  const format = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    format[fromFormat[i]] = seperator[i];
+  }
+
+  for (let j = 0; j < toFormat.length; j++) {
+    if (format.hasOwnProperty(toFormat[j])) {
+      formattedDate.push(format[toFormat[j]]);
+    }
+
+    if (format.hasOwnProperty('YYYY') && toFormat[j] === 'YY') {
+      formattedDate.push(format['YYYY'].slice(2));
+    }
+
+    if (format.hasOwnProperty('YY') && toFormat[j] === 'YYYY') {
+      if (format['YY'] < 30) {
+        formattedDate.push('20'.concat(format['YY']));
+      } else {
+        formattedDate.push('19'.concat(format['YY']));
+      }
```

Comment:
Similarly to the previous comment, the condition and the action for converting 'YY' to 'YYYY' can be moved outside of the loop. This will improve the code structure and performance.